### PR TITLE
samples: adding preconditions to some of the samples

### PR DESCRIPTION
**Pull request by sydney-munro**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: samples/snippets/src/main/java/com/example/storage/object/ReleaseEventBasedHold.java [hunk 1]
```diff
@@ -37,9 +38,14 @@ public static void releaseEventBasedHold(String projectId, String bucketName, St
     // String objectName = "your-object-name";
 
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
-    BlobId blobId = BlobId.of(bucketName, objectName);
+    Blob blob = storage.get(bucketName, objectName);
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/object/ReleaseEventBasedHold.java:41]: Can we expand this to keep blobId and storage.get on separate lines, followed by checking that blob is non-null?

I think it's good to keep the simple `BlobId.of` separate from `storage.get` which performs an RPC to try and be clear about the fact they can be distinct.

File: samples/snippets/src/main/java/com/example/storage/object/ReleaseTemporaryHold.java [hunk 2]
```diff
@@ -38,8 +39,14 @@ public static void releaseTemporaryHold(String projectId, String bucketName, Str
 
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
 
-    BlobId blobId = BlobId.of(bucketName, objectName);
-    storage.update(BlobInfo.newBuilder(blobId).setTemporaryHold(false).build());
+    Blob blob = storage.get(bucketName, objectName);
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/object/ReleaseTemporaryHold.java:42]: ditto

File: samples/snippets/src/main/java/com/example/storage/object/SetEventBasedHold.java [hunk 3]
```diff
@@ -37,9 +38,14 @@ public static void setEventBasedHold(String projectId, String bucketName, String
     // String objectName = "your-object-name";
 
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
-    BlobId blobId = BlobId.of(bucketName, objectName);
+    Blob blob = storage.get(bucketName, objectName);
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/object/SetEventBasedHold.java:41]: ditto

File: samples/snippets/src/main/java/com/example/storage/object/SetTemporaryHold.java [hunk 4]
```diff
@@ -37,9 +38,14 @@ public static void setTemporaryHold(String projectId, String bucketName, String
     // String objectName = "your-object-name";
 
     Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
-    BlobId blobId = BlobId.of(bucketName, objectName);
+    Blob blob = storage.get(bucketName, objectName);
```
Comment (BenWhitehead) [inline samples/snippets/src/main/java/com/example/storage/object/SetTemporaryHold.java:41]: ditto

**Review by BenWhitehead — CHANGES_REQUESTED** [review]

**Review by BenWhitehead — APPROVED** [review]

**Comment by BenWhitehead** [comment]
LGTM, get someone from samples reviewers to give it a once over, then merge away!

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**